module/modmanager/manager.go

```diff
@@ -758,15 +758,35 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	ctxTimeout, cancelFunc := context.WithTimeout(ctx, rutils.GetModuleStartupTimeout(logger))
 	defer cancelFunc()
 
+	slowTicker := time.NewTicker(15 * time.Second)
+	defer slowTicker.Stop()
+	startTime := time.Now()
+
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-slowTicker.C:
+				elapsed := time.Since(startTime).Seconds()
+				logger.Warnf("waiting %q for module to be ready. Elapsed %.2f seconds", m.cfg.Name, elapsed)
+			case <-done:
+				return
+			}
+		}
+	}()
+
 	for {
```

Comment:
After making this change and looking at the code more carefully. I'm not sure why do we have a for loop here. I get the sense that all retrying happens at `resp, err := m.client.Ready(ctxTimeout, req, grpc_retry.WithMax(5000))` and we return either with an error or when is ready. Is there a case where we get a resp that is not ready and with no error?

Reply:
> Is there a case where we get a resp that is not ready and with no error?

IIUC yes there is a case where we get a response but `!resp.Ready`. The module library exposes a `SetReady` method on modules you can use as a module author to explicitly stop your module from returning a true value for `Ready` in the case that you're aware of some hardware still starting up or something. Hence the need for the for loop alongside the gRPC retry.

Reply:
Ohhh TIL! Thank you